profiling: add tips for kubernetes self-managed deployment (#3932)
elastic/observability-docs: docs/en/observability/profiling-self-managed-ops.asciidoc

Replace the old binary and restart the services.

You will find the links to the new binaries in the "Add Data" page, under the "Binary" tab.

[discrete]
[[profiling-self-managed-kubernetes-tips]]
== Kubernetes tips

When deploying the Universal Profiling backend on Kubernetes, there are some best practices to follow.

[discrete]
=== Ingress configuration

If you are using an ingress controller, the connection routing to the collector Service should be configured to use the gRPC protocol.

We provide an `Ingress` resource as part of the Helm chart, and because the ingress can be any implementation,

> you should provide to the chart values the configuration of the ingress, adding the class name and the

Reviewer: > you should provide to the chart values the configuration of the ingress

I'm not sure what you're trying to say here.

Author: The Ingress is managed by the chart, but its configurations need to be passed through from the users, as they are implementation-specific.
This practically means that all the times an ingress is enabled, it must be configured with some values passed in Helm by users.

Reviewer: Thanks, that's very helpful. I made a small recommendation based off of this comment: https://github.com/elastic/observability-docs/pull/3932/files#r1621192289

> We provide an `Ingress` resource as part of the Helm chart, and because the ingress can be any implementation,
> you should provide to the chart values the configuration of the ingress, adding the class name and the
> necessary annotations using the `ingress.annotations` field.

Reviewer: ```suggestion
We provide an `Ingress` resource as part of the Helm chart. Because the ingress can be any implementation,
you must configure the controller with a class name and
any necessary annotations using the `ingress.annotations` field.
```

There is usually no need to customize annotations for this type of service, but the chart provides similar configuration options.

[discrete]
=== Output TLS configuration

You can secure the communication between the Universal Profiling backend and the Elasticsearch cluster by enabling TLS
in the `output.elasticsearch` section of the collector and symbolizer configuration files.

To do so, Kubernetes secrets containing the TLS key pairs should be provisioned in the namespace where the backend is installed.
In case of self-signed certificates, the CA bundle used to validate Elasticsearch's certificates should also be part of the secret.

Create two secrets, one for the collector and one for the symbolizer, with the names `pf-symbolizer-tls-certificate` and `pf-collector-tls-certificate`.
The secrets should contain the following keys:

- `tls.key`: the certificate private key
- `tls.cert`: the certificate public key
- `ca.cert` (optional): the certificate CA bundle

Follow these steps to enable TLS connection from collector/symbolizer to Elasticsearch:

1. Create secrets with the TLS key pairs (omit the `ca.cert` field if you are not using a self-signed CA):

>         kubectl create secret generic pf-collector-tls-certificate --from-file=tls.key=/path/to/key.pem \
>         --from-file=tls.cert=/path/to/cert.pem --from-file=ca.cert=/path/to/ca.crt
> 
>         kubectl create secret generic pf-symbolizer-tls-certificate --from-file=tls.key=/path/to/key.pem \
>         --from-file=tls.cert=/path/to/cert.pem --from-file=ca.cert=/path/to/ca.crt

Reviewer: ```suggestion
[source,terminal]
----
kubectl create secret generic pf-collector-tls-certificate --from-file=tls.key=/path/to/key.pem \
--from-file=tls.cert=/path/to/cert.pem --from-file=ca.cert=/path/to/ca.crt
----

[source,terminal]
----
kubectl create secret generic pf-symbolizer-tls-certificate --from-file=tls.key=/path/to/key.pem \
--from-file=tls.cert=/path/to/cert.pem --from-file=ca.cert=/path/to/ca.crt
----
```

2. Update the collector and symbolizer Helm values files to enable the use of TLS configuration:
+

> [source, yaml, highlight=3..4]

Reviewer: Unfortunately this `highlight` functionality doesn't work in our implementation of asciidoctor.

Author: Bummer... is there anything else we can use to make the diff more evident?

Reviewer: Our docs support [code callouts](https://github.com/elastic/docs?tab=readme-ov-file#callouts). You could add one of those?

Author: I added a small text to clarify what needs to be done 👍🏼 
Thanks for the hint 🙏🏼
